src/devices/machine/pccard_sram.cpp

```diff
@@ -0,0 +1,288 @@
+	PORT_DIPSETTING(   0x00, DEF_STR(No))
+	PORT_DIPNAME(0x02, 0x00, "Battery Low")    PORT_WRITE_LINE_DEVICE_MEMBER(DEVICE_SELF, pccard_sram_device, battery_voltage_2_w)
+	PORT_DIPSETTING(   0x02, DEF_STR(Yes))
+	PORT_DIPSETTING(   0x00, DEF_STR(No))
+	PORT_DIPNAME(0x04, 0x04, "Write Protect")  PORT_WRITE_LINE_DEVICE_MEMBER(DEVICE_SELF, pccard_sram_device, write_protect_w)
+	PORT_DIPSETTING(   0x08, DEF_STR(No))
+	PORT_DIPSETTING(   0x00, DEF_STR(Yes))
+INPUT_PORTS_END
+
+ioport_constructor pccard_sram_device::device_input_ports() const
+{
+	return INPUT_PORTS_NAME(card);
+}
+
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  pccard_sram_device - constructor
+//-------------------------------------------------
+
+pccard_sram_device::pccard_sram_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_memory_interface(mconfig, *this),
+	device_image_interface(mconfig, *this),
+	device_pccard_interface(mconfig, *this),
+	m_card_detect(false),
+	m_switches(*this, "switches")
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void pccard_sram_device::device_start()
+{
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void pccard_sram_device::device_reset()
+{
+	// forward initial state of battery/write protect to slot
+	if (m_card_detect)
+	{
+		m_slot->battery_voltage_1_w(BIT(m_switches->read(), 0));
+		m_slot->battery_voltage_2_w(BIT(m_switches->read(), 1));
+		m_slot->write_protect_w(BIT(m_switches->read(), 2));
+	}
+	else
+	{
+		m_slot->battery_voltage_1_w(0);
+		m_slot->battery_voltage_2_w(0);
+		m_slot->write_protect_w(0);
+	}
+}
+
+//-------------------------------------------------
+//  memory_space_config - configure memory space
+//-------------------------------------------------
+
+device_memory_interface::space_config_vector pccard_sram_device::memory_space_config() const
+{
+	return space_config_vector
+	{
+		std::make_pair(0, &m_memory_space_config),
+		std::make_pair(1, &m_attribute_space_config)
+	};
+}
+
+//-------------------------------------------------
+//  device_pccard_interface
+//-------------------------------------------------
+
+uint16_t pccard_sram_device::read_memory(offs_t offset, uint16_t mem_mask)
+{
+	uint16_t data = 0xffff;
+
+	if (m_card_detect)
+		data = space(0).read_word(offset * 2, mem_mask);
+
+	return data;
+}
+
+void pccard_sram_device::write_memory(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	if (m_card_detect && BIT(m_switches->read(), 3))
+		space(0).write_word(offset * 2, data, mem_mask);
+}
```

Comment:
Please use a `memory_access_specific` initialised on start to improve performance of address space.